2024/YoctoProjectDeveloperDay/practical-sec-tee.md

```diff
@@ -0,0 +1,1228 @@
+  token state:   uninitialized
+(...)
+root@quartz64-a:~# p11 --init-token --label mytoken --so-pin 1234567890
+(...)
+root@quartz64-a:~# p11 --list-slots
+Available slots:
+Slot 0 (0x0): OP-TEE PKCS11 TA - TEE UUID 94e9ab89-4c43-56ea-8b35-45dc07226830
+  token label        : mytoken
+  token manufacturer : Linaro
+```
+
+???
+
+- Time for this slide: 30
+- Idea/goal of this slide: To show our effort to integrate OP-TEE using our port
+  for RK3566 and problems we have met.
+- What to say:
+    - TODO
+- Notes:
+    - Using Rockchip ATF binary and self-compiled OP-TEE OS and TAs binaries via
+      `meta-arm` recipes.
+    - No OP-TEE OS or TAs issues.
+    - PKCS#11 TA and Secure Storage can be initialized and used.
+
+---
+
+# Testing self-ported image
+
+* Running OP-TEE test suite:
+
+```bash
+root@quartz64-a:~# xtest
+(...)
++-----------------------------------------------------
+28130 subtests of which 0 failed
+106 test cases of which 0 failed
+0 test cases were skipped
+TEE test application done!
+```
+
+???
+
+- Time for this slide: 30
+- Idea/goal of this slide: Same as the previous one.
+- What to say:
+    - TODO
+- Notes:
+    - All OP-TEE tests have passed.
+
+---
+
+<!-- Zarhus and Zarhus OS -->
+
+# Contact us
+
+.left-column45[ <img src="/img/zarhus_logo.png" height="150px"
+style="margin-top:-10px; margin-left:200px"> ]
+
+.right-column55[
+
+## [Zarhus OS](https://docs.zarhus.com/)
+
+]
+
+<br>
+<br>
+<br>
+<br>
+<br>
+
+.right-column50[
+
+- <a href="https://3mdeb.com">https://3mdeb.com</a>
+
+- <a href="https://cloud.3mdeb.com/index.php/apps/calendar/appointment/n7T65toSaD9t">Book a call</a>
+
+- <a href="https://newsletter.3mdeb.com/subscription/PW6XnCeK6">
+      Sign up for the newsletter
+    </a>
+  ]
+
+.left-column50[
+
+- <a href="https://www.linkedin.com/company/3mdeb">
+    <img src="/remark-templates/3mdeb-presentation-template/images/linkedin.png"
+      width="24px" style="margin-bottom:-5px; margin-left:-15px"/>
+      linkedin.com/company/3mdeb
+    </a>
+
+- <a href="mailto:contact@3mdeb.com">
+    <img src="/remark-templates/3mdeb-presentation-template/images/email.png"
+      width="24px" style="margin-bottom:-5px; margin-left:-15px"/>
+      contact@3mdeb.com
+  </a>
+
+- <a href="https://www.facebook.com/3mdeb">
+    <img src="/remark-templates/3mdeb-presentation-template/images/facebook.png"
+      width="24px" style="margin-bottom:-5px; margin-left:-15px"/>
+      facebook.com/3mdeb
+  </a>
+
+- <a href="https://twitter.com/3mdeb_com">
+    <img src="/remark-templates/3mdeb-presentation-template/images/x.png"
+      width="24px" style="margin-bottom:-5px; margin-left:-15px"/>
+      @3mdeb_com
+  </a>
+
+  ]
+
+<br>
+<br>
+<br>
+<br>
+<br>
+<br>
+<br>
+<br>
+
+Feel free to contact us if you believe we can help you in any way. We are always
+open to cooperate and discuss.
+
+???
+
+- Time for this slide: 30
+- Idea/goal of this slide: Present Zarhus OS and its features.
```

Comment:
@tym2k1, @artur-rs, could you help me here? What else we can put here. What about our long-term plans?

Reply:
@DaniilKl, we have yet to prepare such a roadmap with the team. But, for now, we can include information like:

* meta-dts, meta-rte and other layers migration to the Zarhus ecosystem
* preparation of secure layer with selected secure features enabled
* focus on expanding the public Zarhus docs with guides and instructions on how to handle Zarhus features
* include support for new platforms - yet to announce the HCL
* we can encourage to follow the zarhus-issues repo: https://github.com/zarhus/zarhus-issues